spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-lattice/src/main/java/org/springframework/cloud/data/module/deployer/lattice/ReceptorTaskModuleDeployer.java

```diff
@@ -0,0 +1,127 @@
+
+	public static final String DOCKER_PATH = "docker:///springcloud/stream-module-launcher";
+
+	private final ReceptorClient receptorClient = new ReceptorClient();
+
+	@Override
+	public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
+		ModuleDeploymentId id = ModuleDeploymentId.fromModuleDefinition(request.getDefinition());
+		String guid = guid(id);
+
+		TaskCreateRequest task = new TaskCreateRequest();
+		task.setLogGuid(guid);
+		task.setTaskGuid(guid);
+		task.setRootfs(DOCKER_PATH);
+
+		RunAction runAction = task.getAction().get("run");
+		runAction.setPath("java");
+		runAction.addArg("-Djava.security.egd=file:/dev/./urandom");
+		runAction.addArg("-jar");
+		runAction.addArg("/module-launcher.jar");
+
+		List<EnvironmentVariable> environmentVariables = new ArrayList<EnvironmentVariable>();
+		Collections.addAll(environmentVariables, task.getEnv());
+		environmentVariables.add(new EnvironmentVariable("MODULES", request.getCoordinates().toString()));
+		environmentVariables.add(new EnvironmentVariable("SPRING_PROFILES_ACTIVE", "cloud"));
+		for (Map.Entry<String, String> entry : request.getDefinition().getParameters().entrySet()) {
+			environmentVariables.add(new EnvironmentVariable(entry.getKey(), entry.getValue()));
+		}
+
+		task.setEnv(environmentVariables.toArray(new EnvironmentVariable[environmentVariables.size()]));
+		task.setMemoryMb(512);
+
+		logger.debug("Desired Task: {}", task);
+		for (EnvironmentVariable e : environmentVariables) {
+			logger.debug("{}={}", e.getName(), e.getValue());
+		}
+
+		receptorClient.createTask(task);
+		return id;
+	}
+
+	/**
+	 * Create a Diego process guid for the given {@link ModuleDeploymentId}.
+	 *
+	 * @param id the module deployment id
+	 * @return string containing a Diego process guid
+	 */
+	// todo: this should be encapsulated in an interface implemented by each SPI implementation
+	private String guid(ModuleDeploymentId id) {
+		return id.toString().replace(".", "_");
+	}
+
+	@Override
+	public void undeploy(ModuleDeploymentId id) {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public ModuleStatus status(ModuleDeploymentId id) {
+		ModuleStatus.Builder builder = ModuleStatus.of(id);
+
+		// todo: if the actual Task is not found, search for the desired Task to verify
+		// that the Task is known to Lattice
+		TaskResponse task = receptorClient.getTask(guid(id));
+
+		Map<String, String> attributes = new HashMap<>();
+		attributes.put("failureReason", task.getFailureReason());
+		attributes.put("result", task.getResult());
+		attributes.put("annotation", task.getAnnotation());
+		attributes.put("completionCallbackUrl", task.getCompletionCallbackUrl());
+		attributes.put("resultFile", task.getResultFile());
+		attributes.put("cellId", task.getCellId());
+		attributes.put("domain", task.getDomain());
+		builder.with(new ReceptorModuleInstanceStatus(task.getTaskGuid(), task.getState(), attributes));
```

Comment:
I'd like to try to keep the single `ReceptorModuleInstanceStatus`. If anything we could parameterize the `State` type and define `TaskState` in addition to `ProcessState`.